src/ds/ptrwrap.h

```diff
@@ -50,7 +50,8 @@ namespace snmalloc
     CBChunk, /*        Chunk                                              */
     CBChunkE, /*       Chunk   (+ platform constraints)                   */
     CBAlloc, /*        Alloc                                              */
-    CBAllocE /*        Alloc   (+ platform constraints)                   */
+    CBAllocE, /*       Alloc   (+ platform constraints)                   */
+    CBAllocEW /*       Alloc,  (+ E, might be "Wild"/forged)              */
```

Comment:
Yes, probably.  I've got a prototype of that on my desk, but I'm not hugely thrilled with it.  I'll open another PR for opinions once it's in reviewable shape.

Reply:
It's not urgent, and certainly not a prerequisite for this landing, but I thought it would make it easier to do things like:
```
template<enum PointerUse Use, bool, IsExported, bool IsWild>
class PointerKind
{
  static constexpr PointerUse use = Use;
  static constexpr is_wild = IsWild;
  static constexpr is_exported = IsExported;
};

enum PointerUse
{
  Arena,
  Chunk,
  Alloc
};

using CBArena = PointerKind<Arena, false, false>;
using CBChunk = PointerKind<Chunk, false, false>; 
using CBChunkE = PointerKind<Chunk, true, false>;
using CBAlloc = PointerKind<Chunk, false, false>; 
using CBAllocE = PointerKind<Chunk, true, false>; 
using CBAllocEW = PointerKind<Chunk, true, true>; 

template<enum PointerUse Use, bool IsExported>
CapPtr<Use, IsExported, false> dewild(CapPtr<Use, IsExported, true> wildPtr)
{
  ...
}
```

That should leave the code that just overloads operations on the concrete types the same, but allow you to write overloads that match some concrete properties but not others.